tests/unit-tests/test_gnn.py

```diff
@@ -2471,6 +2512,448 @@ def check_call_gen_embs(skip_last_self_loop):
+    dataloader1 = GSgnnEdgeDataLoader(ep_data, target_idx, fanout=[-1, -1],
+                                     batch_size=10,
+                                     label_field='label',
+                                     node_feats='feat',
+                                     edge_feats='feat',
+                                     train_task=False,
+                                     remove_target_edge_type=False)
+    pred, labels = edge_mini_batch_gnn_predict(model1, dataloader1, return_label=True)
+    assert pred[("n0", "r1", "n1")].dim() == 2
+    assert pred[("n0", "r1", "n1")].shape[0] == labels[("n0", "r1", "n1")].shape[0]
+
+    # Test case 1: normal case, model uses edge feature, and do full graph inference. 
+    #              This should trigger an assertion error, showing an edge feature supported
+    #              GNN model cannot do full graph inference
+    with assert_raises(AssertionError):
+        do_full_graph_inference(model1, ep_data)
+
+    # Test case 2: normal case, model does not use edge feature, so can do both mini-batch and
+    #              full graph inference
+    model2 = create_rgcn_edge_model(ep_data.g, num_ffn_layers=0)
+    check_edge_prediction(model2, ep_data)
+
+    # Test case 3: normal case, model does not use edge feature, but provide edge feature as inputs.
+    #              This will trigger an assertion error to ask users to use edge_feat_name to create
+    #              a model using edge features.
+    target_idx = {("n0", "r1", "n1"): th.arange(ep_data.g.number_of_edges("r1"))}
+    dataloader2 = GSgnnEdgeDataLoader(ep_data, target_idx, fanout=[-1, -1],
+                                     batch_size=10,
+                                     label_field='label',
+                                     node_feats='feat',
+                                     edge_feats='feat',
+                                     train_task=False,
+                                     remove_target_edge_type=False)
+    with assert_raises(AssertionError):
+        edge_mini_batch_gnn_predict(model2, dataloader2, return_label=True)
+
+    th.distributed.destroy_process_group()
+    dgl.distributed.kvstore.close_kvstore()
+
+def test_rgcn_node_model_forward():
+    # initialize the torch distributed environment
+    th.distributed.init_process_group(backend='gloo',
+                                      init_method='tcp://127.0.0.1:23456',
+                                      rank=0,
+                                      world_size=1)
+    with tempfile.TemporaryDirectory() as tmpdirname:
+        # get the test dummy distributed graph
+        _, part_config = generate_dummy_dist_graph(tmpdirname)
+        np_data = GSgnnData(part_config=part_config,
+                            node_feat_field="feat",
+                            edge_feat_field="feat")
+
+    # Test case 0: normal case, node model uses edge feature, and provides edge feature
```

Comment:
ditto

Reply:
Add another test case, this will trigger an assert error. In addition, this case has been tested in test_embed.py line https://github.com/awslabs/graphstorm/pull/281.